From 6689dd20b553114786b4805dc2c0b1d84fac64cb Mon Sep 17 00:00:00 2001
From: Kazuya Hioki <Kazuya.Hioki@sony.com>
Date: Fri, 25 Jan 2019 15:14:53 +0900
Subject: [PATCH 2/4] Remove mute control from audio library

To minimize the pop noise as possible, the mute control is performed
in the application side. Therefore, remove it from audio library.
---
 .../SPRESENSE/hardware/spresense/1.0.0/libraries/Audio/Audio.cpp  | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/Arduino15/packages/SPRESENSE/hardware/spresense/1.0.0/libraries/Audio/Audio.cpp b/Arduino15/packages/SPRESENSE/hardware/spresense/1.0.0/libraries/Audio/Audio.cpp
index 3e48164..fbe6ee5 100644
--- a/Arduino15/packages/SPRESENSE/hardware/spresense/1.0.0/libraries/Audio/Audio.cpp
+++ b/Arduino15/packages/SPRESENSE/hardware/spresense/1.0.0/libraries/Audio/Audio.cpp
@@ -417,7 +417,9 @@ err_t AudioClass::setReadyMode(void)
       return AUDIOLIB_ECODE_AUDIOCOMMAND_ERROR;
     }
 
-  board_external_amp_mute_control(true);
+  // To minimize the pop noise as possible, the mute control is performed
+  // in the application side. Therefore, remove it from audio library.
+  //board_external_amp_mute_control(true);
 
   destroyStaticPools();
 
@@ -488,7 +490,9 @@ err_t AudioClass::setPlayerMode(uint8_t device, uint8_t sp_drv)
       return AUDIOLIB_ECODE_AUDIOCOMMAND_ERROR;
     }
 
-  board_external_amp_mute_control(false);
+  // To minimize the pop noise as possible, the mute control is performed
+  // in the application side. Therefore, remove it from audio library.
+  //board_external_amp_mute_control(false);
 
   return AUDIOLIB_ECODE_OK;
 }
-- 
2.7.4

